tests/test_sec_approval_workflow.py

```diff
@@ -0,0 +1,225 @@
+    with patch("landoui.landoapi.LandoAPI.request", autospec=True) as api:
+        api.side_effect = LandoAPIDouble()
+        yield api
+
+
+@pytest.fixture
+def authenticated_session(client, monkeypatch):
+    """Simulate a session for a user who has authenticated with Auth0."""
+    # We need to use the session_transaction() method to modify the session
+    # in a way that affects both application code and template code sessions.
+    #
+    # If we only modify the session global then the global session object
+    # will be modified but the template session object will be blank.
+    #
+    # Note: unfortunately the session object in the current process is still
+    # blank so we can't test if landoui.helpers.is_user_authenticated() ==
+    # True.
+    with client.session_transaction() as session:
+        session["id_token"] = "foo_id_token"
+        session["access_token"] = "foo_access_token"
+        session["userinfo"] = {"picture": ""}
+
+        # These two values need to be present in the session for the OIDC
+        # auth library to evaluate the session as fresh and authenticated.
+        session["id_token_jwt"] = "foo_jwt"
+        session["last_authenticated"] = time.time()
+
+
+@pytest.fixture
+def anonymous_session(client):
+    """Simulate a session for an unauthenticated (anonymous) user."""
+    # We need to use the session_transaction() method to modify the session
+    # in a way that affects both application code and template code sessions.
+    #
+    # If we only modify the session global then the global session object
+    # will be modified but the template session object will be blank.
+    #
+    # Note: unfortunately the session object in the current process is still
+    # blank so we can't test if landoui.helpers.is_user_authenticated() ==
+    # False.
+    with client.session_transaction() as session:
+        assert not session.keys()  # Check just to be sure.
+
+
+def test_view_stack_not_logged_in(client, anonymous_session, apidouble):
+    # Basic happy-path test for an anonymous user.
+    rv = client.get("/D1/")
+    assert rv.status_code == 200
+
+
+def test_view_stack_logged_in(client, authenticated_session):
+    # Basic happy-path test for a logged in user.
+    rv = client.get("/D1/")
+    assert rv.status_code == 200
+
+
+def test_sec_approval_workflow_mark_hidden_if_revision_is_public(
+    client, authenticated_session
+):
+    rv = client.get("/D1/")
+    assert rv.status_code == 200
+    assert (
+        b"<!-- This revision is subject to the sec-approval workflow -->"
+        not in rv.data
+    )
+
+
+def test_sec_approval_workflow_mark_shown_if_revision_is_secure(
+    client, authenticated_session, apidouble
+):
+    apidouble.side_effect.stack_response["revisions"][0]["is_secure"] = True
+    rv = client.get("/D1/")
+    assert rv.status_code == 200
+    assert (
+        b"<!-- This revision is subject to the sec-approval workflow -->" in
```

Comment:
This type of testing where we're checking the html output could get pretty fragile IMO. I worry the cost of repeatedly updating these greatly outweighs the value they bring.